matchms/similarity/BaseSimilarity.py

```diff
@@ -28,87 +30,127 @@ class BaseSimilarity:
+            For example: return np.asarray(score, dtype=self.score_datatype)
         """
         raise NotImplementedError
 
-    def matrix(self, references: Sequence[SpectrumType],
-               queries: Sequence[SpectrumType],
-               is_symmetric: bool = False,
-               mask_indices: COOIndex = None,
-               ) -> np.ndarray:
+    def matrix(
+            self,
+            references: Sequence[SpectrumType],
+            queries: Sequence[SpectrumType],
+            is_symmetric: bool = False,
+            mask_indices: COOIndex = None) -> np.ndarray:
         """
         Compute a dense similarity matrix for all pairs of reference and query spectra.
 
+        Use this method when filtering is either required (with a provided mask) or when working
+        with moderately sized datasets. If no filtering is needed and you don't have a mask, simply
+        pass an empty tuple for `score_filters` and use this method.
+
         Parameters
         ----------
         references:
             Collection of reference spectra.
         queries:
             Collection of query spectra.
-        mask_indices:
-            Indices to calculate scores for the rest is set to 0.
         is_symmetric:
-            Indicates if the similarity matrix is symmetric (e.g., for all-vs-all comparisons).
-            When True, only the upper triangle of the matrix is computed and then mirrored,
-            which can reduce computation time.
+            If True, indicates that the similarity matrix is symmetric (i.e., references and queries are the same).
+            Only the upper triangle is computed and then mirrored. Defaults to False.
+        mask_indices:
+            A COOIndex instance specifying which pairs to compute.
+            If provided, only the specified index pairs will be computed (others remain zero).
+            This is helpful when a previous score already filters out many pairs, reducing computation time and memory footprint. 
         """
         if mask_indices is None:
             return self._matrix_without_mask_with_filter(references, queries, is_symmetric=is_symmetric)
         return self._matrix_with_mask(references, queries,
                                       mask_indices=mask_indices, is_symmetric=is_symmetric)
 
-    def sparse_array(self, references: Sequence[SpectrumType], queries: Sequence[SpectrumType],
-                     mask_indices: COOIndex = None, is_symmetric = False) -> COOMatrix:
+    def sparse_array(
+            self,
+            references: Sequence[SpectrumType],
+            queries: Sequence[SpectrumType],
+            mask_indices: COOIndex = None,
+            is_symmetric = False) -> COOMatrix:
+        """
+        Compute a sparse array (in COO format) of similarity scores.
+
+        Use this method if you expect heavy filtering (i.e. many scores are dropped) or
+        if you want to compute scores only for a selected set of index pairs.
```

Comment:
```suggestion
        if you want to compute scores only for a selected set of index pairs. By using sparse_array() this can reduce the memory footprint if many scores are dropped. 
```